openmls/src/group/mls_group/app_ephemeral.rs

```diff
@@ -0,0 +1,200 @@
+use super::proposal_store::{ProposalQueue, QueuedAppEphemeralProposal};
+
+use crate::extensions::ComponentId;
+
+use std::collections::BTreeSet;
+
+impl ProposalQueue {
+    /// Return an iterator over the [`QueuedAppEphemeralProposal`]s in the proposal queue,
+    /// in the order they appear in the commit.
+    pub fn app_ephemeral_proposals_for_component_id(
+        &self,
+        component_id: ComponentId,
+    ) -> impl Iterator<Item = QueuedAppEphemeralProposal<'_>> {
+        self.app_ephemeral_proposals()
+            .filter(move |p| p.app_ephemeral_proposal().component_id == component_id)
+    }
+    /// Return the list of all [`ComponentId`]s available across all
+    /// [`QueuedAppEphemeralProposal`]s in the proposal queue.
+    pub fn unique_component_ids_for_app_ephemeral(&self) -> impl Iterator<Item = ComponentId> {
+        let ids: BTreeSet<_> = self
+            .app_ephemeral_proposals()
+            .map(|p| p.app_ephemeral_proposal().component_id)
+            .collect();
+
+        ids.into_iter()
```

Comment:
If you collect anyway, there's no benefit in returning an iterator. Let's just return the list.